content/blog/integrating-istio-and-spire.md

```diff
@@ -0,0 +1,463 @@
+
+You can get IPs of these services by following command:
+
+```shellsession
+k8s-spiffe-integ-master-7j7fh-m67q9:~ kubectl get svc -n istio-system
+NAME                   TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)                                      AGE
+istio-ingressgateway   LoadBalancer   10.105.191.32   172.16.17.5   15021:30189/TCP,80:30392/TCP,443:30566/TCP   32d
+istiod                 ClusterIP      10.101.27.65    <none>        15010/TCP,15012/TCP,443/TCP,15014/TCP        32d
+kiali                  LoadBalancer   10.103.14.197   172.16.17.6   20001:32116/TCP,9090:31950/TCP               32d
+prometheus             ClusterIP      10.98.101.102   <none>        9090/TCP                                     32d
+```
+
+![](/img/manual_proxy.png)
+
+**Format:** `http://{external ip};`
+
+**Note:** Your kiali service might be of ClusterIP type, so to get the external IP for this service, you first need to edit the service type to LoadBalancer.
+
+* Use the following command to edit the service, then edit the service type.
+
+  ```shellsession
+  kubectl edit svc kiali -n istio-system
+  ```
+
+  Edit the service type **{spec: {type:LoadBalancer}}** as shown below
+
+  ![](/img/service_edit.png)
+
+**5﻿.6**  Set the ingress IP and ports:
+
+```shellsession
+export INGRESS_NAME=istio-ingressgateway
+
+export INGRESS_NS=istio-system
+
+export INGRESS_HOST=$(kubectl -n "$INGRESS_NS" get service "$INGRESS_NAME" -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+
+export INGRESS_PORT=$(kubectl -n "$INGRESS_NS" get service "$INGRESS_NAME" -o jsonpath='{.spec.ports[?(@.name=="http2")].port}')
+```
+
+**5.7**  Export and Set GATEWAY_URL: 
+
+```shellsession
+export GATEWAY_URL=$INGRESS_HOST:$INGRESS_PORT
+```
+
+```shellsession
+k8s-spiffe-integ-master-7j7fh-m67q9:~ echo "$GATEWAY_URL"
+172.16.17.5:80
+```
+
+Curl into productpage using gateway URL using following command.
+
+```shellsession
+k8s-spiffe-integ-master-7j7fh-m67q9:~ curl -v  http://$GATEWAY_URL/productpage
+* Uses proxy env variable no_proxy == 'localhost,127.0.0.1,10.96.0.1,172.16.5.41,172.16.5.42,172.16.5.43,172.16.5.44,172.16.5.45,172.16.5.46,172.16.5.40,glhc-caas.glhc-hpe.local,.glhc-hpe.local,glhc-caas.customer.hpe.net,172.16.17.20,172.16.17.21,172.16.17.22,172.16.5.47,gl-pulpnode.glhc-hpe.local,gl-pulpnode,10.96.0.1,10.192.0.0/12,10.96.0.0/12,.svc,.cluster.local,.default.svc,.customer.hpe.net,172.16.17.23,172.16.17.30,gl-cp-gw-node2.glhc-hpe.local,gl-cp-gw-node1.glhc-hpe.local,172.16.17.50'
+* Uses proxy env variable http_proxy == 'http://172.16.0.250:8080'
+*   Trying 172.16.0.250:8080...
+* TCP_NODELAY set
+* Connected to 172.16.0.250 (172.16.0.250) port 8080 (#0)
+> GET http://172.16.17.5:80/productpage HTTP/1.1
+> Host: 172.16.17.5
+> User-Agent: curl/7.66.0
+> Accept: */*
+> Proxy-Connection: Keep-Alive
+>
+```
+
+You can generate traffic on product page by just reaching out to shown http URL. 
```

Comment:
add the article "the" prior to "product page". Also add "the" prior to "shown http URL."